test/integration/unregistered_user_authentication_test.rb

```diff
@@ -0,0 +1,72 @@
+require 'test_helper'
+
+class UnregisteredUserAuthenticationTest < ActionDispatch::IntegrationTest
+  def logged_in_user
+    User.create(username: "user", password: "password")
+    visit '/login'
+    fill_in "Username", with: "user"
+    fill_in "Password", with: "password"
+    click_button "Login"
+  end
+
+  def create_user
+    User.create(username: "user", password: "password")
+  end
+
+  test "unregistered user can register" do
+    visit '/'
+
+    click_link "Create Account"
+
+    assert new_user_path, current_path
+
+    fill_in "Username", with: "user"
+    fill_in "Password", with: "password"
+    click_button "Create Account"
+
+    assert '/dashboard', current_path
+
+    assert page.has_content?("Logged in as User")
+    assert page.has_content?("User's Dashboard")
+  end
+
+  test "unregistered user can login" do
+    create_user
+    user = User.find_by(username: 'user')
+    visit '/'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.